Assets/Treasure/TDK/Runtime/Connect/TDK.Connect.cs

```diff
@@ -117,7 +117,7 @@ public async Task SetChainId(ChainId chainId)
             
             var thirdwebService = TDKServiceLocator.GetService<TDKThirdwebService>();
             await thirdwebService.SwitchNetwork(GetChainIdAsInt());
-            // TODO recreate user session?
+            // TODO does user session need to be recreated when chain is switched?
```

Comment:
@alecananian i think i asked on the context of the treasure launcher and you mentioned we should make sure the session exists on the new chain. Is that also the case for the TDK?

I ask in part because logic to start a new session on chain switch didnt seem to be implemented before, and i havent run into any issues (as far as i know), so im not 100% sure if we need to do something here

Reply:
Yes it's required for any consumers of the TDK - transactions won't work unless there's an active session on the chain. It's my understanding that the Thirdweb SDK v5 doesn't have the concept of setting an active chain, though. It would be great if we can get rid of this `SetChainId` function and instead use the same patterns as Thirdweb, passing the chain in as params.

Reply:
The new `Thirdweb.SmartWallet` does in fact have a `SwitchNetwork` method (used by TDKThirdwebService), which makes it easy to reuse the same `ActiveWallet` for all chains.

From what I see the chain id is mostly relevant in 3 places:
- for thirdwebService.ActiveWallet (smart wallet needs to be aware of the chain, internally)
- for the ValidateUserSession call (looking into this)
- for setting the `X-Chain-Id` header in requests with TDK.API

I will look a bit deeper and sync with you

Reply:
As discussed in Slack, the `CreateSessionKey` method uses the chain ID set on the active `SmartWallet`, so we can keep the current functionality. We can also rely on partners to validate and create sessions when they're needed to avoid eager session creation, which costs gas funds.